cli/util/upload_bin.go

```diff
@@ -224,21 +224,21 @@ func fetchLatestMissingBlockIndex(ctx context.Context, p *pool.Pool, containerID
 					objectIDs, err = neofs.ObjectSearch(ctx, p, priv, containerID.String(), prm)
 					return err
 				}, maxRetries)
-				results[i] = searchResult{startIndex: startIndex, endIndex: endIndex, numOIDs: len(objectIDs), err: err}
-			}(i, startIndex, endIndex)
+				results[i] = searchResult{batchStartIndex: startIndex / uploadBatchSize, index: startIndex, numOIDs: len(objectIDs), err: err}
+			}(i, startIndex)
 		}
 		wg.Wait()
 
 		for i := len(results) - 1; i >= 0; i-- {
 			if results[i].err != nil {
-				return 0, fmt.Errorf("blocks search failed for batch with indexes from %d to %d: %w", results[i].startIndex, results[i].endIndex-1, results[i].err)
+				return 0, fmt.Errorf("blocks search failed for batch starting with index %d, block index %d: %w", results[i].batchStartIndex, results[i].index, results[i].err)
 			}
 			if results[i].numOIDs == 0 {
 				emptyBatchFound = true
 				continue
 			}
 			if emptyBatchFound || (batch == numBatches && i == len(results)-1) {
-				return results[i].startIndex, nil
+				return results[i].batchStartIndex * uploadBatchSize, nil
```

Comment:
Replace it with `return results[i].startIndex / uploadBatchSize * uploadBatch, nil`. It will give proper results for both single-object search and multy-object search. Check it.